src/rust/aggregator/src/data/user_history.rs

```diff
@@ -285,51 +265,32 @@ impl Data for UserHistory {
         }
         if !swap_events.is_empty() {
             for x in swap_events {
-                sqlx::query!(
-                    r#"
-                        INSERT INTO aggregator.aggregated_events VALUES (
-                            $1, $2
-                        );
-                    "#,
-                    x.txn_version,
-                    x.event_idx,
-                )
-                .execute(&mut transaction as &mut PgConnection)
-                .await
-                .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
+                mark_as_aggregated(&mut transaction, &x.txn_version, &x.event_idx).await?;
             }
         }
-        for x in &change_events {
-            sqlx::query!(
-                r#"
-                    UPDATE aggregator.user_history
-                    SET remaining_size = $1, last_updated_at = $4
-                    WHERE order_id = $2 AND market_id = $3;
-                "#,
-                x.new_size,
-                x.order_id,
-                x.market_id,
-                x.time,
-            )
-            .execute(&mut transaction as &mut PgConnection)
-            .await
-            .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
-        }
-        if !change_events.is_empty() {
-            for x in change_events {
-                sqlx::query!(
-                    r#"
-                        INSERT INTO aggregator.aggregated_events VALUES (
-                            $1, $2
-                        );
-                    "#,
-                    x.txn_version,
-                    x.event_idx,
-                )
-                .execute(&mut transaction as &mut PgConnection)
-                .await
-                .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
-            }
+        for i in 0..fill_events.len().max(change_events.len()) {
+            match (fill_events.get(i), change_events.get(i)) {
+                (Some(fill), Some(change)) => {
+                    if fill.txn_version < change.txn_version || (fill.txn_version == change.txn_version && fill.event_idx < change.event_idx) {
+                        aggregate_fill(&mut transaction, &fill.size, &fill.maker_order_id, &fill.market_id, &fill.time).await?;
```

Comment:
This is incorrect. You might aggregate say, 8 fill events before a change size event. Just because the change size event is adjacent to the fill event doesn't mean that it gets aggregated immediately afterward.

Reply:
Done